lib/Tomcat/ModjkTest.pm

```diff
@@ -0,0 +1,111 @@
+
+    # install and configure tomcat in console
+    $self->select_serial_terminal();
+
+    record_info('install and configure apache2 and apache2-mod_jk connector Setup');
+    zypper_call('in apache2 apache2-mod_jk');
+
+    $self->conn_apache2_tomcat();
+    $self->load_jk_module();
+}
+
+sub func_conn_apache2_tomcat() {
+    my $self = shift;
+    # Connection from apache2 to tomcat: Functionality test
+    $self->select_serial_terminal();
+    systemctl('stop apache2');
+    systemctl('stop tomcat');
+    assert_script_run(
+        "echo  \"\$(cat <<EOF
+JkMount /*       ajp13
+JkMount /*.jsp   ajp13
+JkMount /*.xhtml ajp13
+EOF
+        )\"  >> /etc/apache2/conf.d/jk.conf", timeout => 180
+    );
+    systemctl('start apache2');
+    systemctl('status apache2');
+
+    systemctl('start tomcat');
+    systemctl('status tomcat');
+}
+
+sub conn_apache2_tomcat() {
+    # Connection from apache2 to tomcat: tomcat part
+    systemctl('stop tomcat');
+
+    # Define an AJP 1.3 Connector on IPv6 localhost port 8009
+    my $ajp_connector = qq(    <Connector protocol="AJP/1.3" address="::1" port="8009" redirectPort="8443" \/>);
+    assert_script_run qq(sed -i '/<\!-- Define an AJP 1.3 Connector on port 8009 -->/ a $ajp_connector\' /etc/tomcat/server.xml);
+
+    # Define a worker "ajp13" which listens behind the above AJP 1.3 Connector address and port
+    assert_script_run('cp /usr/share/doc/packages/apache2-mod_jk/workers.properties /etc/tomcat/workers.properties');
+    assert_script_run(
+        "echo \"\$(cat <<EOF
+worker.list=ajp13
+worker.ajp13.reference=worker.template
+worker.ajp13.host=localhost
+worker.ajp13.port=8009
+EOF
+        )\" >> /etc/tomcat/workers.properties", timeout => 180
+    );
+
+    # Connection from apache2 to tomcat: apache2 part
+    systemctl('stop apache2');
+    assert_script_run('cp -ai /usr/share/doc/packages/apache2-mod_jk/jk.conf /etc/apache2/conf.d');
+    if (is_sle('=15') or is_sle('<12-sp4')) {
+        record_soft_failure 'boo#1167896';
+        assert_script_run(
+            "echo \"\$(cat <<EOF
+JkShmFile /var/log/tomcat/jk-runtime-status
+EOF
+        )\" >> /etc/apache2/conf.d/jk.conf", timeout => 180
+        );
+        assert_script_run("sed -i 's|servlets-examples|examples/servlets|g' /etc/apache2/conf.d/jk.conf");
+        assert_script_run("sed -i 's|jsp-examples|examples/jsp|g' /etc/apache2/conf.d/jk.conf");
+    }
+}
+
+sub load_jk_module() {
+    # Include mod_jk into the apache2 list of modules to load
+    assert_script_run('cp -a /etc/sysconfig/apache2 sysconfig.apache2-1-no.jk');
+    # Configure inclusion of mod_jk into apache2 service
+    assert_script_run('a2enmod jk');
+    if ((script_run 'diff sysconfig.apache2-1-no.jk /etc/sysconfig/apache2') != 1) {
+        die "failed to appended jk to the APACHE_MODULES line";
+    }
+    systemctl('start apache2');
+    validate_script_output("apachectl -M | grep jk_module",                        sub { ".*jk_module.*" });
```

Comment:
As I am not very familiar with the topic, is it normal to get here these kind of messages? : http://10.161.229.147/tests/1142#step/tomcat/264 
`AH00557: httpd-prefork: apr_sockaddr_info_get() failed for susetest`

Reply:
It's apache being apache, there's a `Server localhost` line missing in the config file, but it would not cause any trouble